internal/component/prometheus/relabel/relabel.go

```diff
@@ -47,22 +55,40 @@ type Arguments struct {
 	// The relabelling rules to apply to each metric before it's forwarded.
 	MetricRelabelConfigs []*alloy_relabel.Config `alloy:"rule,block,optional"`
 
-	// Cache size to use for LRU cache.
-	CacheSize int `alloy:"max_cache_size,attr,optional"`
+	// DEPRECATED Use type = inmemory and cache_size field.
+	InMemoryCacheSizeDeprecated int `alloy:"max_cache_size,attr,optional"`
+
+	// Cache backend configuration.
+	CacheConfig cache.CacheConfig `alloy:"cache,block,optional"`
 }
 
 // SetToDefault implements syntax.Defaulter.
 func (arg *Arguments) SetToDefault() {
 	*arg = Arguments{
-		CacheSize: 100_000,
+		CacheConfig: cache.CacheConfig{
+			Backend: cache.InMemory,
+			InMemory: cache.InMemoryCacheConfig{
+				CacheSize: 100_100,
+			},
+		},
 	}
 }
 
 // Validate implements syntax.Validator.
 func (arg *Arguments) Validate() error {
-	if arg.CacheSize <= 0 {
-		return fmt.Errorf("max_cache_size must be greater than 0 and is %d", arg.CacheSize)
+	switch arg.CacheConfig.Backend {
+	case cache.InMemory:
+		if arg.CacheConfig.InMemory.CacheSize <= 0 {
+			return fmt.Errorf("cache_size must be greater than 0 and is %d", arg.CacheConfig.InMemory.CacheSize)
+		}
+	case cache.Memcached:
+		// todo
```

Comment:
Need to determine what to include here, maybe move this to the service/cache package ?